code/datums/components/hallucinations.dm

```diff
@@ -0,0 +1,537 @@
+		if(image_list ~= src.image_list && src.target_list ~= target_list) //this is the same hallucination, just update timeout and prob, time
+			if(timeout == -1)
+				src.ttl = timeout
+			else if(src.ttl != -1)
+				src.ttl = world.time + timeout SECONDS //reset timeout
+			src.range = range
+			src.image_prob = image_prob
+			src.image_time = image_time
+			src.override = override
+			return TRUE //no duplicate
+		else
+			return FALSE //create a new hallucination
+
+//#########################################################
+//                    SUPPORTING CAST
+//#########################################################
+
+
+/datum/hallucinated_sound
+	///The sound file to play
+	var/path
+	///Max number of times to play it
+	var/max_count
+	///Min number of times to play it
+	var/min_count
+	///Delay between each play
+	var/delay
+	///Pitch to play it at
+	var/pitch
+
+	New(path, min_count = 1, max_count = 1, delay = 0, pitch = 1)
+		..()
+		src.path = path
+		src.min_count = min_count
+		src.max_count = max_count
+		src.delay = delay
+		src.pitch = pitch
+
+	///Play the sound to a mob from a location
+	proc/play(var/mob/mob, var/atom/location)
+		SPAWN(0)
+			for (var/i = 1 to rand(src.min_count, src.max_count))
+				mob.playsound_local(location, src.path, 100, 1, pitch = src.pitch)
+				sleep(src.delay)
+
+/obj/fake_attacker
+	icon = null
+	icon_state = null
+	var/fake_icon = 'icons/misc/critter.dmi'
+	var/fake_icon_state = ""
+	name = ""
+	desc = ""
+	density = 0
+	anchored = ANCHORED
+	opacity = 0
+	var/mob/living/carbon/human/my_target = null
+	var/weapon_name = null
+	///Does this hallucination constantly whack you
+	var/should_attack = TRUE
+	event_handler_flags = USE_FLUID_ENTER
+
+	proc/get_name()
+		return src.fake_icon_state
+
+	pig
+		fake_icon = 'icons/effects/hallucinations.dmi'
+		fake_icon_state = "pig"
+		get_name()
+			return pick("pig", "DAT FUKKEN PIG")
+	spider
+		fake_icon_state = "big_spide"
+		get_name()
+			return pick("giant black widow", "aw look a spider", "OH FUCK A SPIDER")
+	slime
+		fake_icon = 'icons/effects/hallucinations.dmi'
+		fake_icon_state = "slime"
+		get_name()
+			return pick("red slime", "some gooey thing", "ANGRY CRIMSON POO")
+	shambler
+		fake_icon = 'icons/effects/hallucinations.dmi'
+		fake_icon_state = "shambler"
+		get_name()
+			return pick("shambler", "strange creature", "OH GOD WHAT THE FUCK IS THAT THING?")
+	legworm
+		fake_icon_state = "legworm"
+	handspider
+		fake_icon_state = "handspider"
+
+	eyespider
+		fake_icon_state = "eyespider"
+	buttcrab
+		fake_icon_state = "buttcrab"
+		should_attack = FALSE
+	bat
+		fake_icon_state = "bat"
+		get_name()
+			return pick("bat", "batty", "the roundest possible bat", "the giant bat that makes all of the rules")
+	snake
+		fake_icon_state = "rattlesnake"
+		get_name()
+			return pick("snek", "WHY DID IT HAVE TO BE SNAKES?!", "rattlesnake", "OH SHIT A SNAKE")
+
```

Comment:
more whitespace nitpicks- not sure what the Standard is here since we don't usually deal with long lists of stubby types, but I'd be fine with either putting a blank line below each type, or only doing it for types with the proc override (ie ling critters can stay as a clump). just make it consistent one way or another, or a secret 3rd way